### ISSUE TITLE
Connect-Rsc dosen't work in RubrikSecurityCloud 1.7

### ISSUE THREAD

**Issue by SahFari** [rubrikinc/rubrik-powershell-sdk#131]
I installed RubrikSecurityCloud 1.7 on a lab server this morning, this after that have uninstalled older versions of RubrikSecurityCloud on the same server.

I couldn’t connect to RSC when I used PS 5. It worked in PS7!

Did many tests, uninstalled, and reinstalled RubrikSecurityCloud 1.7 several times, rebooted the server. It didn’t work. 

Checked it on my own computer. It worked there. Checked if I had other versions of RubrikSecurityCloud installed. I had 11 versions in PS 7 and 6 versions in PS 5!

Get-InstalledModule RubrikSecurityCloud -AllVersions

I start testing some more on my lab server. 

- Uninstalled RubrikSecurityCloud 1.7.
- Installed RubrikSecurityCloud 1.5. Could connect to RSC. 
- Installed (updated) RubrikSecurityCloud to 1.7. Could connect to RSC.
- Uninstalled all versions and installed 1.7. Couldn’t connect.
- Installed 1.5 & 1.7. Could connect to RSC. 

Did this 3 times! 
This is what I got when I only had installed 1.7 and couldn’t connect:
 
DEBUG: [Connect_Rsc] RscBasePSCmdlet.cs:154:BeginProcessing()
  Cmdlet Inputs:
DEBUG: [Connect_Rsc] Connect-Rsc.cs:303:_Connect()
  Before Auth: AuthState=NOT_AUTHORIZED, PSVar=
DEBUG: [Connect_Rsc] RscSessionInfo.cs:59:.ctor()
  Sending GQL Query:
  query GetRscVersion{
  deploymentVersion
  
  }
connect-rsc : One or more errors occurred.
At line:1 char:1
+ connect-rsc -Debug
+ ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : AuthenticationError: (:) [Connect-Rsc], AggregateException
    + FullyQualifiedErrorId : UnableToAuthenticateToRubrikSecurityCloud,RubrikSecurityCloud.PowerShell.Cmdlets.Connect_Rsc

**Comment by kacummings** [comment]
I have the same issue. Been messing with it all day until I saw this and tried 1.5. It works for me, whereas 1.7 fails auth with the same error.

**Comment by SahFari** [comment]
@kacummings I ended up removing 1.7 from PS 5. I got the same issue after a reboot yesterday! 
Did 2 more tests yesterday, there I removed 1.7, and reinstalled it once again. Rebooted the server after each test.

After a reboot 1.7 can’t connect to RSC in PS 5. 

For the 5:th time I removed 1.7, leaving 1.5 in PS 5. Could connect to RSC after a reboot. 

I logged on to that server today. I could connect and disconnect to RSC via PS 5.
1.7 is still working on PS 7.

**Comment by rynardtspies** [comment]
Thank you for bringing this issue to our attention.

We have identified a library conflict between .NET Framework and one of the GraphQL libraries that is causing the problem in Windows PowerShell 5.1. Our team is actively collaborating with Microsoft to resolve this issue as soon as possible.

In the meantime, we recommend using PowerShell 7, which is not affected by this conflict.

We apologize for any inconvenience this may cause and appreciate your understanding.

Best regards,
Rynardt Spies

Technical Specialist
Rubrik Professional Services

**Comment by kacummings** [comment]
Thank you. Late yesterday I looked into the GitHub repo for this module and found that someone else commented the same issue a few days ago. I was able to work around it using their suggestion- by removing v1.7 of the module, then installing v1.5.

That being said, I still haven't been able to find how to list all active/running jobs by object type (backup, archive, recovery) with RSC module as I had done in the past with the Rubrik PS module. I did find a way to create a token for the service user and login to the cluster locally via the old Rubrik module. For the time being that will work, as long as the token continues to work.

We push the running tasks to a dashboard and refresh every couple of minutes. It just helps us keep an eye on potential issues. If you have any suggestions or examples for pull the same information with RSC module, please let me know.

***@***.***

Thank you!

Kyle Cummings
Cornerstone Information Systems
***@***.***
812-330-4361
________________________________

**Comment by SahFari** [comment]
@kacummings 

Service Accounts needs to be created in RSC, download the .json file to a location and then run a script 
`Set-RscServiceAccountFile -InputFilePath "C:\location_of_the_Json_file` to create a "token" and then delete the .json file.

I can see running tasks by running this:
```
Get-RscEventSeries | 
Where-Object {$_.LastActivityStatus -eq 'RUNNING'} | 
Format-Table ObjectType, ObjectId, LastActivityStatus, ActivitySeriesId, ClusterUuid, IsCancelable, IsPolarisEventSeries

```
Object types 
`Get-RscEventSeries | Select-Object ObjectType -Unique`

**Comment by rynardtspies** [comment]
This issue has been resolved in [Version 1.8](https://github.com/rubrikinc/rubrik-powershell-sdk/releases/tag/Version_1.8)
